### PR TITLE
[build] Add license attribution for dlc images

### PR DESCRIPTION
**Pull request by jeet4320**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: mxnet/training/docker/1.8.0/py3/cu110/Dockerfile.gpu [hunk 1]
```diff
@@ -209,6 +210,12 @@ ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/de
 
 RUN chmod +x /usr/local/bin/deep_learning_container.py
 
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://asimov-utility-packages.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR}
```
Comment (saimidu) [inline mxnet/training/docker/1.8.0/py3/cu110/Dockerfile.gpu:217]: Are we intentionally not deleting the downloaded zip file and the folder to which the zip file is extracted, or can we at least delete the zip file?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 2]
```diff
@@ -479,3 +482,56 @@ def _assert_artifact_free(output, stray_artifacts):
...
+    And upload source of third party packages to S3 bucket.
+    """
+    THIRD_PARTY_SOURCE_CODE_BUCKET = "asimov-package-licenses"
+    THIRD_PARTY_SOURCE_CODE_BUCKET_PATH = "third_party_source_code"
+    file = "THIRD_PARTY_SOURCE_CODE_URLS"
+    container_name = get_container_name("oss_compliance", image)
+    context = Context()
+    local_repo_path = get_repository_local_path()
+    start_container(container_name, image, context)
+    output = run_cmd_on_container(container_name, context, "/usr/local/bin/testOSSCompliance /root")
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_pre_release.py:500]: I don't see `output` being used anywhere further down in this script.

File: test/dlc_tests/sanity/test_pre_release.py [hunk 3]
```diff
@@ -479,3 +482,56 @@ def _assert_artifact_free(output, stray_artifacts):
...
+    start_container(container_name, image, context)
+    output = run_cmd_on_container(container_name, context, "/usr/local/bin/testOSSCompliance")
+
+    try:
+        run(f"docker cp {container_name}:/root/{file} {os.path.join(local_repo_path, file)}")
+    finally:
+        run(f"docker rm -f {container_name}", hide=True)
+
+    context.run(
+        f"ls {local_repo_path}", timeout=60
```
[the remark this answers is not in the document]
Comment (jeet4320) [reply test/dlc_tests/sanity/test_pre_release.py:508]: I had added this just to check files on CB file system, will remove

File: test/dlc_tests/sanity/test_pre_release.py [hunk 4]
```diff
@@ -479,3 +482,56 @@ def _assert_artifact_free(output, stray_artifacts):
         assert not re.search(
             artifact, output.stdout
         ), f"Matched {artifact} in {output.stdout} while running {output.command}"
+
+@pytest.mark.integration("oss_compliance")
+@pytest.mark.model("N/A")
+def test_oss_compliance(image):
```
[the remark this answers is not in the document]
Comment (jeet4320) [reply test/dlc_tests/sanity/test_pre_release.py:488]: yes normal user need to have aws credentials. It is as per other tests in pre_release, is there any marker need to be added?

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.cpu [hunk 5]
```diff
@@ -104,6 +105,15 @@ ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/de
 
 RUN chmod +x /usr/local/bin/deep_learning_container.py
 
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://asimov-utility-packages.s3.amazonaws.com/oss_compliance.zip \
```
Comment (jeet4320) [inline mxnet/inference/docker/1.8.0/py3/Dockerfile.cpu:109]: will change its name

**Comment by jeet4320** [comment]
@saimidu this is ready for review now

**Comment by jeet4320** [comment]
@saimidu lets merge this if you are good

File: test/testrunner.py [hunk 6]
```diff
@@ -278,6 +278,7 @@ def main():
     benchmark_mode = "benchmark" in test_type or is_benchmark_dev_context()
     specific_test_type = re.sub("benchmark-", "", test_type) if "benchmark" in test_type else test_type
     test_path = os.path.join("benchmark", specific_test_type) if benchmark_mode else specific_test_type
+    test_path = f"{os.path.join(test_path)}/test_pre_release.py::test_oss_compliance"
```
Comment (saimidu) [inline test/testrunner.py:281]: Will this be reverted after the review?

File: src/config/build_config.py [hunk 7]
```diff
@@ -5,9 +5,9 @@
 # Do remember to revert it back to False before merging any PR (including NEURON dedicated PR)
 ENABLE_NEURON_MODE = False
 # Frameworks for which you want to disable both builds and tests
-DISABLE_FRAMEWORK_TESTS = []
+DISABLE_FRAMEWORK_TESTS = ["tensorflow", "huggingface_tensorflow", "huggingface_pytorch"]
 # Disable new builds or build without datetime tag
-DISABLE_DATETIME_TAG = False
+DISABLE_DATETIME_TAG = True
```
Comment (saimidu) [inline src/config/build_config.py:10]: Reviewer Note: To be reverted after approval.

File: src/config/test_config.py [hunk 8]
```diff
@@ -3,8 +3,8 @@
...
 # Disable the test codebuild jobs to be run
 DISABLE_SANITY_TESTS = False
-DISABLE_SAGEMAKER_TESTS = False
-DISABLE_ECS_TESTS = False
-DISABLE_EKS_TESTS = False
-DISABLE_EC2_TESTS = False
+DISABLE_SAGEMAKER_TESTS = True
+DISABLE_ECS_TESTS = True
+DISABLE_EKS_TESTS = True
+DISABLE_EC2_TESTS = True
```
Comment (saimidu) [inline src/config/test_config.py:9]: Reviewer Note: To be reverted after approval.

**Review by saimidu — COMMENTED** [review]
APPROVED.

**Comment by saimidu** [comment]
Please resolve merge conflicts.

File: test/testrunner.py (hunk 6 above)
Comment (jeet4320) [reply test/testrunner.py:281]: yes

**Review by saimidu — APPROVED** [review]

**Pull request merged**